File: php/testcode/fileinclusion_036.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_fi_intval_file_number
function fileinclusion036(BenchmarkRequest $req): BenchmarkResponse {
    $id = (int) $req->param('id');
    require_once __DIR__ . '/views/' . $id . '.php'; // vuln-code-snippet safe-line php_fi_intval_file_number
    return BenchmarkResponse::ok('Rendered');
}
// vuln-code-snippet end php_fi_intval_file_number

File: ruby/testcode/fileinclusion_036.rb
require_relative 'shared'

PLUGINS = { 'pdf' => 'plugins/pdf.rb', 'csv' => 'plugins/csv.rb' }.freeze

# vuln-code-snippet start ruby_fi_plugin_registry
def handler(req)
  load(PLUGINS.fetch(req.param('key'))) # vuln-code-snippet safe-line ruby_fi_plugin_registry
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_fi_plugin_registry
